File: 3sum_closest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
    let closestSum = Number.MAX_VALUE;
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            for (let k = j + 1; k < nums.length; k++) {
                if (Math.abs(target - closestSum) > 
                    Math.abs(target - (nums[i] + nums[j] + nums[k])))
                        closestSum = (nums[i] + nums[j] + nums[k]);

            }

        }
    }

    return closestSum;
};

File: 3sum_closest.py
import sys
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        closestSum = sys.maxsize
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                for k in range(j + 1, len(nums)):
                    if(abs(target-closestSum) > abs(target- (nums[i]+ nums[j] + nums[k]))):
                        closestSum = (nums[i] + nums[j] + nums[k])
                    return closestSum

        
